pkg/proxy/keepalive/keepalive_darwin.go

```diff
@@ -0,0 +1,52 @@
+//go:build darwin
+// +build darwin
+
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package keepalive
+
+import (
+	"syscall"
+
+	"github.com/pingcap/TiProxy/lib/config"
+)
+
+const (
+	// missing in older darwin
+	_TCP_KEEPINTVL = 0x101
+	_TCP_KEEPCNT   = 0x102
+)
+
+func setKeepalive(syscn syscall.RawConn, cfg config.KeepAlive) error {
+	var serr error
+	return errors.Collect(ErrKeepAlive, serr, syscn.Control(func(fd uintptr) {
+		serr = syscall.SetsockoptInt(int(fd), syscall.SOL_SOCKET, syscall.SO_KEEPALIVE, 1)
+		if serr != nil {
+			return
+		}
+		serr = syscall.SetsockoptInt(int(fd), syscall.IPPROTO_TCP, syscall.TCP_KEEPALIVE, int(cfg.Idle.Seconds()))
+		if serr != nil {
+			return
+		}
+		serr = syscall.SetsockoptInt(int(fd), syscall.IPPROTO_TCP, _TCP_KEEPCNT, cfg.Cnt)
```

Comment:
What will happen if cnt==0? It's 0 by default.
I think if cnt==0/idle==0/intvl==0, do not set it. Or set a default value to them.